h2o-algos/src/main/java/hex/ensemble/MetalearnerProvider.java

```diff
@@ -1,10 +1,13 @@
 package hex.ensemble;
 
 import hex.Model.Parameters;
+import water.api.schemas3.ModelParametersSchemaV3;
 
-public interface MetalearnerProvider<M extends Metalearner> {
+public interface MetalearnerProvider<M extends Metalearner, MPS extends ModelParametersSchemaV3> {
```

Comment:
I'm a bit allergic to those dependency loops:
- model API depends on schema API
- schema API depends on model API

and even worse when it's used in generic parameters because it will propagate those dependency loops like a virus…

At the minimum, I would remove it from the generic signature to avoid propagation: we abuse generics beyond necessity sometimes in this project to the point where we often struggle against the compiler to get things right, which should NOT be the purpose of generics (I blame the design of generics itself for that): we should generify code when it becomes useful instead of complexifying it from the beginning (I take my part of responsibility in this…).

Here, we just need `ModelParametersSchemaV3` (technically, we even just need `Schema`, which is in `core` module, that makes it a more tolerable loop): I don't see any reason to ask for more.